src/containers/Signup/Signup.js

```diff
@@ -24,13 +25,19 @@ class Signup extends Component {
    */
   constructor() {
     super();
-    this.state = {
+
+
+    this.initialState = {
       email: '',
       username: '',
       password: '',
       confirmPassword: ''
     };
 
+    this.state = {
+      ...this.initialState
+    };
```

Comment:
😄 fancy but not need ... 
you can just add the values to the state object directly... 
benefits: removes unnecessary declaration and reduces the lines of code in this file